heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java

```diff
@@ -219,4 +222,142 @@ public static int allocateNewContainer(ArrayList<Container> containers, Resource
     return componentsToScaleUp;
   }
 
+  /**
+   * Identifies the resources reclaimed by the components that will be scaled down
+   *
+   * @return Total resources reclaimed
+   */
+  public static Resource getScaleDownResource(TopologyAPI.Topology topology, Map<String, Integer>
+      componentChanges, Resource defaultInstanceResources) {
+    double cpu = 0;
+    long ram = 0;
+    long disk = 0;
+    Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
+    Map<String, Integer> componentsToScaleDown = PackingUtils.getComponentsToScaleDown(
+        componentChanges);
+    for (String component : componentsToScaleDown.keySet()) {
+      int parallelismChange = -componentChanges.get(component);
+      cpu += parallelismChange * defaultInstanceResources.getCpu();
+      disk += parallelismChange * defaultInstanceResources.getDisk();
+      if (ramMap.containsKey(component)) {
+        ram += parallelismChange * ramMap.get(component);
+      } else {
+        ram += parallelismChange * defaultInstanceResources.getRam();
+      }
+    }
+    return new Resource(cpu, ram, disk);
+  }
+
+  /**
+   * Identifies the resources needed for the components that will be scaled up
+   *
+   * @return Total resources needed
+   */
+  public static Resource getScaleUpResource(TopologyAPI.Topology topology, Map<String, Integer>
+      componentChanges, Resource defaultInstanceResources) {
+    double cpu = 0;
+    long ram = 0;
+    long disk = 0;
+    Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
+    Map<String, Integer> componentsToScaleUp = PackingUtils.getComponentsToScaleUp(
+        componentChanges);
+    for (String component : componentsToScaleUp.keySet()) {
+      int parallelismChange = componentChanges.get(component);
+      cpu += parallelismChange * defaultInstanceResources.getCpu();
+      disk += parallelismChange * defaultInstanceResources.getDisk();
+      if (ramMap.containsKey(component)) {
+        ram += parallelismChange * ramMap.get(component);
+      } else {
+        ram += parallelismChange * defaultInstanceResources.getRam();
+      }
+    }
+    return new Resource(cpu, ram, disk);
+  }
+
+  /**
+   * Computes the additional resources needed to accommodate a scale up and down operation
```

Comment:
We could change the name to imply the operation (e.g. `subtractAbsolute`?) or just document that subtract will never go negative. Since it's a mathematical operation between two `Resource` items it seems fitting to have this as a method on `Resource` IMO. It's more OO than static helpers.